index.bs

```diff
@@ -500,6 +500,37 @@ request they are associated with.
 
 Issue(139): Clarify scoping/uniqueness of request IDs.
 
+Type Key Backwards Compatibility
+--------------------------------
+
+As messages are modified or extended over time, certain rules must be followed
+to maintain backwards compatibiilty with agents that understand older versions
+of messages.
+
+1.  If a required field is added to or removed from a message (either to/from the
+message directly or indirectly through the field of a field), a new type key
+must be assigned to the message.  Is is effectively a new message and must not
+be sent unless the receiving agent is known to understand the new type key.
+
+1.  If an optional field is added to a message (either to the message directly
+or indirectly through the field of a field), the type key may remain unchanged
+if the behavior of older receiving agents that do not understand the added field
+is compatible with newer sendering agents that do understand the added field.
+Otherwise, a new type key must be assigned.
+
+
+1.  If an optional field is removed from a message (either from the message
+directly or indirectly through the field of a field), the type key may remain
+unchanged if the behavior of newer receiving agents that do not understand the
+removed field is compatible with older sending agents that do understand the
+removed field.  Otherwise, a new type key must be assigned.  
+
+1. Required fields may not be added or removed from array-based messages (either
```

Comment:
It's okay to add or remove a field from a nested dictionary if you leave the array as-is and follow the rules above.  So maybe just drop the part in parenthesis?

Reply:
I removed the parenthesis.